### PR TITLE
Bug 1779218 - [PerfCompare] Calculate delta, delta percentage, magnitude

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**

### CODE REVIEW

**Review by kimberlythegeek — APPROVED** [review]
Ran the code locally, and confirmed that the values for delta, delta %, and magnitude were the same as those calculated in perfherder.

Thanks!

File: treeherder/webapp/api/perf_compare_utils.py [hunk 1]
```diff
@@ -148,3 +148,22 @@ def is_improvement(lower_is_better, base_avg_value, new_avg_value):
...
+def get_delta_value(new_avg_value, base_avg_value):
+    return new_avg_value - base_avg_value
+
+
+def get_delta_percentage(delta_value, base_avg_value):
+    return get_percentage(delta_value, base_avg_value)
+
+
+def new_is_better(delta_value, lower_is_better):
+    return (lower_is_better and delta_value < 0) or (not lower_is_better and delta_value > 0)
```
Comment (eng-esther) [inline treeherder/webapp/api/perf_compare_utils.py:165]: [suggestion] might be worth adding some comments here why this logic is like so

File: treeherder/webapp/api/perf_compare_utils.py [hunk 2]
```diff
@@ -148,3 +148,22 @@ def is_improvement(lower_is_better, base_avg_value, new_avg_value):
...
+
+def get_delta_value(new_avg_value, base_avg_value):
+    return new_avg_value - base_avg_value
+
+
+def get_delta_percentage(delta_value, base_avg_value):
+    return get_percentage(delta_value, base_avg_value)
+
+
+def new_is_better(delta_value, lower_is_better):
```
Comment (eng-esther) [inline treeherder/webapp/api/perf_compare_utils.py:164]: my js brain reads the function name as 
```suggestion
def is_new_better(delta_value, lower_is_better):
```

File: tests/webapp/api/test_performance_data_api.py [hunk 3]
```diff
@@ -683,6 +683,40 @@ def test_perfcompare_results_multiple_runs(
 
     option_collection_map = perf_compare_utils.get_option_collection_map()
 
+    first_row = dict()
+    first_row['base_avg_value'] = statistics.mean(sig1_values)
+    first_row['new_avg_value'] = statistics.mean(sig2_values)
+    first_row['is_improvement'] = perf_compare_utils.is_improvement(
```
Comment (eng-esther) [inline tests/webapp/api/test_performance_data_api.py:689]: are you also going to add changes for `one_run`

File: tests/webapp/api/test_performance_data_api.py [hunk 4]
```diff
@@ -855,6 +897,23 @@ def test_perfcompare_results_with_only_one_run(
 
     option_collection_map = perf_compare_utils.get_option_collection_map()
 
+    response = dict()
```
Comment (eng-esther) [inline tests/webapp/api/test_performance_data_api.py:900]: [question] would using like so be better ?
```suggestion
    response = {}
```

**Review by eng-esther — APPROVED** [review]

File: treeherder/webapp/api/perf_compare_utils.py (hunk 1 above)
Comment (beatrice-acasandrei) [reply treeherder/webapp/api/perf_compare_utils.py]: I don't have more insight for this logic unfortunately, only some comments from the current Compare View code. Left a simplistic comment though.

File: tests/webapp/api/test_performance_data_api.py (hunk 4 above)
Comment (beatrice-acasandrei) [reply tests/webapp/api/test_performance_data_api.py]: I have changed it to your suggestion, I don't think it makes much of a difference though.

File: tests/webapp/api/test_performance_data_api.py (hunk 3 above)
Comment (beatrice-acasandrei) [reply tests/webapp/api/test_performance_data_api.py]: I've done the same thing for the other test as well.

**Review by eng-esther — APPROVED** [review]
lgtm

**Pull request merged**